routes/applications.ts

```diff
@@ -0,0 +1,22 @@
+const express = require("express");
+const { SUPERUSER } = require("../constants/roles");
+const authenticate = require("../middlewares/authenticate");
+const authorizeRoles = require("../middlewares/authorizeRoles");
+const applications = require("../controllers/applications");
+const authorizeOwnOrSuperUser = require("../middlewares/authorizeOwnOrSuperUser");
+const applicationValidator = require("../middlewares/validators/application");
+
+const router = express.Router();
+
+router.get("/", authenticate, authorizeOwnOrSuperUser, applications.getAllOrUserApplication);
```

Comment:
this should have a validator that validates the query params we are passing here.